src/shared_types.rs

```diff
@@ -76,6 +78,13 @@ impl Identifier {
     pub fn as_str(&self) -> &str {
         &self.0
     }
+
+    /// Return the u64 hash value of self.
+    pub fn hash_self(&self) -> u64 {
```

Comment:
I might just call this method `hash`?

Reply:
I think this is already taken by the Hash trait :D

Reply:
the same name can exist on a type and a trait, that's no problem.

Reply:
... I think?

Reply:
Only one  way to find out

Reply:
Hm, nope. It would result in a recursive call, see the body of the function.

Reply:
try:

```rust
    pub fn hash(&self) -> u64 {
        let mut hasher = DefaultHasher::new();
        Hash::hash(self, &mut hasher);
        hasher.finish()
    }
```